File: src/main/java/org/example/XMLToJSON.java
package org.example;

import org.w3c.dom.*;
import javax.xml.parsers.*;
import org.json.JSONArray;
import org.json.JSONObject;
import java.io.*;

public class XMLToJSON{
    public static void main(String[] args) {
        try {
            // XML-Datei einlesen
            File inputFile = new File("book_store.xml");
            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
            DocumentBuilder builder = factory.newDocumentBuilder();
            Document doc = builder.parse(inputFile);
            doc.getDocumentElement().normalize();  // Normalisieren der XML-Daten

            // JSON-Objekt und Array erstellen
            JSONObject bookstoreJson = new JSONObject();
            JSONArray booksArray = new JSONArray();

            // Alle Bücher durchgehen und konvertieren
            NodeList bookList = doc.getElementsByTagName("book");
            for (int i = 0; i < bookList.getLength(); i++) {
                Node bookNode = bookList.item(i);
                if (bookNode.getNodeType() == Node.ELEMENT_NODE) {
                    Element bookElement = (Element) bookNode;

                    // Buchdaten extrahieren
                    String category = bookElement.getAttribute("category");
                    String title = bookElement.getElementsByTagName("title").item(0).getTextContent();
                    String author = bookElement.getElementsByTagName("author").item(0).getTextContent();
                    String year = bookElement.getElementsByTagName("year").item(0).getTextContent();
                    String price = bookElement.getElementsByTagName("price").item(0).getTextContent();

                    // Buch in ein JSONObject konvertieren
                    JSONObject bookData = new JSONObject();
                    bookData.put("category", category);
                    bookData.put("title", title);
                    bookData.put("author", author);
                    bookData.put("year", year);
                    bookData.put("price", price);

                    // Buch zur Liste hinzufügen
                    booksArray.put(bookData);
                }
            }

            // JSON-Daten in das Haupt-JSONObject einfügen
            bookstoreJson.put("bookstore", booksArray);

            // JSON-Datei speichern
            FileWriter file = new FileWriter("bookstore.json");
            file.write(bookstoreJson.toString(4));  // Mit Einrückungen speichern
            file.close();

            System.out.println("JSON-Datei wurde erstellt!");

        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: xmltojson.py
import xml.etree.ElementTree as ET
import json

# XML-Datei einlesen
tree = ET.parse('book_store.xml')  # Hier wird die XML-Datei geladen
root = tree.getroot()  # Wurzelelement des XML-Dokuments holen (<bookstore>)


# XML-Daten in ein Dictionary konvertieren
def bookstore_xml_dic(root):
    bookstore = {"bookstore": []}  # Start mit einem leeren 'bookstore' Array

    # Iteriere durch die Kategorien von Büchern
    for category in root.findall('book'):
        # Für jedes Buch im XML, erstelle ein Dictionary
        book_data = {
            "category": category.get('category'),
            "title": category.find('title').text,
            "author": category.find('author').text,
            "year": category.find('year').text,
            "price": category.find('price').text
        }
        bookstore['bookstore'].append(book_data)  # Das Buch zur Liste hinzufügen

    return bookstore


# convert XML-Daten in Python-Dictionary
bookstore_data = bookstore_xml_dic(root)

# Speichern der Daten in einer JSON-Datei
with open('bookstore.json', 'w') as json_file:
    json.dump(bookstore_data, json_file, indent=4)


print(json.dumps(bookstore_data, indent=4))

print("JSON-Datei wurde erstellt!")
